# 01. Importing libraries

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os 


# 02. Creating a string of the path 

In [5]:
path = r'C:\Users\rbhte\Documents\Achievement_4_Monami\Instacart Basket Analysis'
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']
df_orders = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', '4.3_orders_products', 'orders.csv'), usecols = vars_list, index_col = False)

In [3]:
print(df_orders.head())

   order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0  


In [46]:
df_prods = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', '4.3_orders_products', 'products.csv'), index_col = False) 
print ('  Printing 1st 20 rows ')
print (' ------------------')
print (df_prods.head(20))
print ('  Printing last 35 rows ')
print (' -------------------')
print (df_prods.tail(35))

  Printing 1st 20 rows 
 ------------------
    product_id                                       product_name  aisle_id  \
0            1                         Chocolate Sandwich Cookies        61   
1            2                                   All-Seasons Salt       104   
2            3               Robust Golden Unsweetened Oolong Tea        94   
3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4            5                          Green Chile Anytime Sauce         5   
5            6                                       Dry Nose Oil        11   
6            7                     Pure Coconut Water With Orange        98   
7            8                  Cut Russet Potatoes Steam N' Mash       116   
8            9                  Light Strawberry Blueberry Yogurt       120   
9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
10          11                                  Peach Mango Juice        31   
11      

In [47]:
print (' Column names')
print ('-------------------------------------')
col = df_prods.columns
print (col)

 Column names
-------------------------------------
Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')


In [48]:
print (' Total number of rows and columns')
print ('---------------------------------')
row_col = df_prods.shape
print (row_col)

 Total number of rows and columns
---------------------------------
(49693, 5)


In [49]:
print (' Column names and info')
print ('-------------------------------------')
info = df_prods.info()
print (info) 

 Column names and info
-------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49693 entries, 0 to 49692
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   product_id     49693 non-null  int64  
 1   product_name   49677 non-null  object 
 2   aisle_id       49693 non-null  int64  
 3   department_id  49693 non-null  int64  
 4   prices         49693 non-null  float64
dtypes: float64(1), int64(3), object(1)
memory usage: 1.9+ MB
None


Maximum value of the “aisle_id” column : 49693 

Data type of the “department_id” column : int64